types/Cargo.toml

```diff
@@ -27,6 +27,7 @@ serde = { version = "1.0.106", default-features = false }
 thiserror = "1.0"
 tiny-keccak = { version = "2.0.2", default-features = false, features = ["sha3"] }
 
+channel = { path = "../common/channel", version = "0.1.0" }
```

Comment:
oh wow, this isn't good.... we're going to have to rethink what we're doing here cc: @zekun000